File: java-solutions/src/main/java/solutions/_004_MedianOfTwoSortedArrays.java
package solutions;

import java.util.Arrays;

public class _004_MedianOfTwoSortedArrays {
    public double findMedianSortedArrays(int[] nums1, int[] nums2) {
        int[] all = new int[nums1.length + nums2.length];
        int i = 0;
        int j = 0;
        int k = 0;
        // Traverse both array
        while (i < nums1.length && j < nums2.length) {
            // Check if current element of first
            // array is smaller than current element
            // of second array. If yes, store first
            // array element and increment first array
            // index. Otherwise do same with second array
            if (nums1[i] < nums2[j]) {
                all[k++] = nums1[i++];
            } else {
                all[k++] = nums2[j++];
            }
        }
        // Store remaining elements of first array
        while (i < nums1.length)
            all[k++] = nums1[i++];

        // Store remaining elements of second array
        while (j < nums2.length)
            all[k++] = nums2[j++];

        double result = 0.0f;
        if (all.length % 2 == 0) {
            // even
            result = (all[all.length / 2 - 1] + all[all.length / 2]) / 2.0f;
        } else {
            // odd
            result = all[all.length / 2];
        }

        System.out.println(Arrays.toString(all));

        return result;
    }
}

File: kotlin-solutions/src/main/kotlin/solutions/_004_MedianOfTwoSortedArrays.kt
package solutions

class _004_MedianOfTwoSortedArrays {
    fun findMedianSortedArrays(nums1: IntArray, nums2: IntArray): Double {
        val arr = IntArray(nums1.size + nums2.size)
        var i = 0
        var j = 0
        var k = 0

        while (i < nums1.size && j < nums2.size) {
            if (nums1[i] < nums2[j]) {
                arr[k++] = nums1[i++]
            } else {
                arr[k++] = nums2[j++]
            }
        }

        while (i < nums1.size) {
            arr[k++] = nums1[i++]
        }

        while (j < nums2.size) {
            arr[k++] = nums2[j++]
        }

        return when {
            arr.size % 2 == 0 -> {
                (arr[arr.size / 2 - 1] + arr[arr.size / 2]) / 2.0
            }
            else -> {
                arr[arr.size / 2] * 1.0
            }
        }
    }
}